nautobot/extras/models/approvals.py

```diff
@@ -175,6 +189,19 @@ def save(self, *args, **kwargs):
             *args: positional arguments
             **kwargs: keyword arguments
         """
+        # Check of the object under review's content type matches the parent approval workflow's content type
+        if self.object_under_review_content_type != self.approval_workflow.model_content_type:
+            raise ValidationError(
+                f"The content type {self.object_under_review_content_type} of "
+                f"the object under review does not match the content type {self.approval_workflow.model_content_type} of the parent approval workflow."
+            )
+
+        # TODO need to check of the object fits the model_constraints of the approval workflow
+        if not self.user_name or self.user_name == "Undefined":
```

Comment:
Not sure if this check is needed - if we change `self.user` (if that's even allowed?) I would expect `self.user_name` to automatically change as well even if it were previously set?